src/JsonApi/Serializer/ItemNormalizer.php

```diff
@@ -222,10 +223,6 @@ protected function normalizeRelation(ApiProperty $propertyMetadata, ?object $rel
         if (null !== $relatedObject) {
             $iri = $this->iriConverter->getIriFromResource($relatedObject);
             $context['iri'] = $iri;
-
-            if (isset($context['resources'])) {
```

Comment:
as $tagCollector can be null we need to keep this call for backward compatibility

Reply:
Ah, you mean if someone relies on `$context['resources']` to be populated for other means?

Should such code be marked somehow, so they are removed in later major versions? Or is there no intention to remove in the future?

Reply:
I'm not sure if it's really needed to enforce the use of the service as it worked for simpler cases without it

Reply:
I think I misunderstood the intention first, but I believe I get it know: You propose that I keep the old code (wrapped in `if(!$this->tagCollector)` and in return remove the `TagCollector.php` implementation from this PR?

I can do that. But I don't think it resolves the breaking tests. As soon as I use a custom tag collector, the output is not the same anymore and hence the old tests break. So I need to isolate the service configuration with the custom tag collector from the old tests.

Test were fine before adding the latest commit https://github.com/api-platform/core/pull/5758/commits/a404dd7fd2b6d05057ecfea1c7b79687182edd0a with the customized tag collector.

Reply:
keep these 3 lines if the TagCollector is null as its optional and we don't want to change these. If the tagCollector is defined, then we use it instead.

Reply:
Done with commit https://github.com/api-platform/core/pull/5758/commits/c935d8d165f34983f7801f06d5dfe29ac02a0250